localstack/services/lambda_/invocation/docker_runtime_executor.py

```diff
@@ -349,6 +357,19 @@ def start(self, env_vars: dict[str, str]) -> None:
                 container_config.ports = PortMappings()
             container_config.ports.add(config.LAMBDA_INIT_DELVE_PORT, config.LAMBDA_INIT_DELVE_PORT)
 
+        if (
+            self.function_version.config.layers
+            and not config.LAMBDA_PREBUILD_IMAGES
+            and self.function_version.config.package_type == PackageType.Zip
+        ):
+            # avoid chmod on mounted code paths
+            hot_reloading_env = container_config.env_vars.get(HOT_RELOADING_ENV_VARIABLE, "")
+            if "/opt" not in hot_reloading_env:
```

Comment:
sidenote: if we are going to support advanced globbing in the future (see https://github.com/localstack/localstack/issues/7466), we should keep in mind that this check might not catch it. This would lead to chmod being applied on the host (or failure for read-only mounts).

Reply:
Good point! Not sure how we will implement globbing in the end, but definitely something to consider!